src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPostBuildStatus.java

```diff
@@ -43,6 +42,11 @@ public Revision decorateRevisionToBuild(GitSCM scm, Run<?, ?> run, GitClient git
             listener.getLogger().println("Injector could not be found while creating build status");
             return rev;
         }
+        // If the class was serialized before upgrading to 1.1.2
+        if (refNameExtractorFactory == null) {
```

Comment:
Hmm, thinking a bit more about this: so if we change the implementation of `BitbucketRefNameExtractorFactory` in future versions, we're gonna have the same problem again. :thinking: 
Is it worth then to _always_ just create a fresh instance here every time? It's not a heavy-weight object, just a very simple and lightweight POJO, so let's not serialize it with the rest of this.